cpg-library/src/main/java/de/fraunhofer/aisec/cpg/frontends/cpp/StatementHandler.java

```diff
@@ -271,7 +271,9 @@ private ForEachStatement handleForEachStatement(CPPASTRangeBasedForStatement ctx
     ForEachStatement statement = NodeBuilder.newForEachStatement(ctx.getRawSignature());
     lang.getScopeManager().enterScope(statement);
 
-    Declaration var = this.lang.getDeclarationHandler().handle(ctx.getDeclaration());
+    Declaration decl = this.lang.getDeclarationHandler().handle(ctx.getDeclaration());
+    DeclarationStatement var = new DeclarationStatement();
```

Comment:
It would be better to use the `NodeBuilder` pattern. We discourage directly creating these objects, as the node builder sets some "required" parameters. And yes I am aware how your python code works, let's keep that as a future issue ;)